src/garage/torch/algos/pearl.py

```diff
@@ -1,6 +1,635 @@
+        num_train_tasks (int): Number of tasks for training.
+        num_test_tasks (int): Number of tasks for testing.
+        latent_dim (int): Size of latent context vector.
+        encoder_hidden_sizes (list[int]): Output dimension of dense layer(s) of
+            the context encoder.
+        policy_lr (float): Policy learning rate.
+        qf_lr (float): Q-function learning rate.
+        vf_lr (float): Value function learning rate.
+        context_lr (float): Inference network learning rate.
+        policy_mean_reg_coeff (float): Policy mean regulation weight.
+        policy_std_reg_coeff (float): Policy std regulation weight.
+        policy_pre_activation_coeff (float): Policy pre-activation weight.
+        soft_target_tau (float): Interpolation parameter for doing the
+            soft target update.
+        kl_lambda (float): KL lambda value.
+        optimizer_class (callable): Type of optimizer for training networks.
+        recurrent (bool): Whether or not context encoder is recurrent.
+        use_information_bottleneck (bool): False means latent context is
+            deterministic.
+        use_next_obs_in_context (bool): Whether or not to use next observation
+            in distinguishing between tasks.
+        meta_batch_size (int): Meta batch size.
+        num_steps_per_epoch (int): Number of iterations per epoch.
+        num_initial_steps (int): Number of transitions obtained per task before
+            training.
+        num_tasks_sample (int): Number of random tasks to obtain data for each
+            iteration.
+        num_steps_prior (int): Number of transitions to obtain per task with
+            z ~ prior.
+        num_steps_posterior (int): Number of transitions to obtain per task
+            with z ~ posterior.
+        num_extra_rl_steps_posterior (int): Number of additional transitions
+            to obtain per task with z ~ posterior that are only used to train
+            the policy and NOT the encoder.
+        num_steps_per_eval (int): Number of transitions to evaluate on.
+        batch_size (int): Number of transitions in RL batch.
+        embedding_batch_size (int): Number of transitions in context batch.
+        embedding_mini_batch_size (int): Number of transitions in mini context
+            batch; should be same as embedding_batch_size for non-recurrent
+            encoder.
+        max_path_length (int): Maximum path length.
+        discount (float): RL discount factor.
+        replay_buffer_size (int): Maximum samples in replay buffer.
+        reward_scale (int): Reward scale.
+        update_post_train (int): How often to resample context when obtaining
+            data during training (in trajectories).
+        eval_deterministic (bool): Whether to make policy deterministic during
+            evaluation.
+
+    """
+
+    # pylint: disable=too-many-statements
+    def __init__(self,
+                 env,
+                 policy_class,
```

Comment:
i think that both of these classes should have default args, to save most people time from looking them up (most people won't change these)